account_asset_management_xls/report/account_asset_report_xls.py

```diff
@@ -655,7 +655,7 @@ def _active_report(self, _p, _xs, data, objects, wb):
         cr.execute(
             "SELECT id FROM account_asset "
             "WHERE (purchase_date <= %(period_start)s"
-            "   OR date_start <= %(period_start)s)"
+            "   OR date_start <= %(period_end)s)"
```

Comment:
I think this is the equivalent from a later version of this module: https://github.com/OCA/account-financial-tools/blob/10.0/account_asset_management_xls/report/account_asset_report_xls.py#L431-L432

If we follow it, we should drop the clause on the removal date. Does our intervention with the purchase date mean that we need to check for assets with a purchase date between these two dates as well?

Reply:
OK, thanks for clarifying. There are still some differences between the query here and that one. Please explain why `purchase_date` is still compared to `period_start` when `date_start` is compared to `period_end`, and why `date_remove` is compared to `period_end` while it is compared to date_start in the 10.0 version.

Reply:
In another ticket, we needed to take the purchase_date into account with the reports, so that's why this filter is not available upstreams.

By filtering on date_start <= period_end, we take the assets that are active in these period (because the start date is before the end of the period). For example: previously, we took a report of a period (jan 2020 - dec 2020) ; and an asset with the start date of may 2020 was filtered out, but that one should be on the active assets tab).

Reply:
OK, but that means that with this current query, we don't select
* assets with a purchase date between period_start and period_end, if the asset itself is only started after period_end
* assets with a start date before period_start but with a removal date before period_end.
Is this correct? Asking because at least the latter are selected in the query in V10 that you referred to.

Reply:
Yes, that's right
- assets with a purchase date between period_start and period_end, if the asset itself is only started after period_end > those doesnt have any deprecations, so they are only shown on the new acquisition tab
- assets with a start date before period_end but with a removal date also before period_end > those are shown on the removal tab

Made these use cases with Els:
Usecase 1:
```
Ik koop een laptop op 30-01-2020, ik schrijf af vanaf 01-01-2021
Ik maak een asset rapport van 01-01-2020 t/m 31-12-2020
Laptop staat enkel op new aquisitions
```

Usecase 2:
```
Ik koop een laptop op 01-12-2019, ik schrijf meteen af
Ik maak een asset rapport van 01-01-2020 t/m 31-12-2020
Laptop staat op active assets
```

Usecase 3:
```
Ik koop een laptop op 02-01-2020, ik schrijf deze volledig af in 2020
Ik maak een asset rapport van 01-01-2020 t/m 31-12-2020
Laptop staat op active assets
Ik klik op remove (met een datum in 2020)
Laptop staat op removal assets
```

Reply:
Perfect, thanks!